python/cudf/cudf/tests/test_joining.py

```diff
@@ -1159,3 +1159,110 @@ def test_typecast_on_join_categorical(dtype_l, dtype_r):
 
     got = gdf_l.merge(gdf_r, on="join_col", how="inner")
     assert_eq(expect, got, check_dtype=False)
+
+
+def test_typecast_on_join_indexes():
```

Comment:
May be worth running the operation against Pandas if possible to make sure our typecasting behavior is consistent with their expectations as opposed to just hardcoding golden data. Will make triaging subtle changes in behavior easier.

Reply:
I can do this for some of the tests. The reason I hardcoded the data in all of these is because we decided at some point to break with Pandas in some cases. The justification was inconsistencies like the following:

```
>>> df1 = pd.DataFrame.from_dict({'A':[1,2,3]}, dtype='int32')
>>> df2 = pd.DataFrame.from_dict({'A':[1,2,3]}, dtype='category')
>>> df1['A'].dtype
dtype('int32')
>>> df2['A'].dtype
CategoricalDtype(categories=[1, 2, 3], ordered=False)
>>> df1.merge(df2, on='A')['A'].dtype
dtype('O')
>>> df2.merge(df1, on='A')['A'].dtype
dtype('int64')
```
I can certainly update the tests where we actually match Pandas behavior, but it won't work for all of them. At some point, it actually bloats the test code more to 'undo' some of the things that pandas does and create an `expect` that is actually what we expect.

Reply:
Got it. Lets just stick with our expected behavior then. Thanks for the explanation!